packages/@webex/plugin-meetings/src/reachability/index.ts

```diff
@@ -395,16 +409,207 @@ export default class Reachability {
+      .filter((result) => result[protocol].result === 'reachable')
+      .map((result) => result[protocol].latencyInMilliseconds);
+
+    if (values.length === 0) {
+      return {
+        min: -1,
+        max: -1,
+        average: -1,
+      };
+    }
+
+    return {
+      min: Math.min(...values),
+      max: Math.max(...values),
+      average: mean(values),
+    };
+  }
+
+  /**
+   * Sends a metric with all the statistics about how long reachability took
+   *
+   * @returns {void}
+   */
+  protected async sendMetric() {
+    const results = [];
+
+    Object.values(this.clusterReachability).forEach((clusterReachability) => {
+      results.push({
+        ...clusterReachability.getResult(),
+        isVideoMesh: clusterReachability.isVideoMesh,
+      });
+    });
+
+    const stats = {
+      vmn: {
+        udp: this.getStatistics(results, 'udp', true),
+      },
+      public: {
+        udp: this.getStatistics(results, 'udp', false),
+        tcp: this.getStatistics(results, 'tcp', false),
+        xtls: this.getStatistics(results, 'xtls', false),
+      },
+    };
+    Metrics.sendBehavioralMetric(BEHAVIORAL_METRICS.REACHABILITY_COMPLETED, stats);
+  }
+
+  /**
+   * Starts all the timers used for various timeouts
+   *
+   * @returns {void}
+   */
+  private startTimers() {
+    this.vmnTimer = setTimeout(() => {
+      this.vmnTimer = undefined;
+      // if we are only missing VMN results, then we don't want to wait for them any longer
+      // as they are likely to fail if users are not on corporate network
+      if (this.areAllPublicClusterResultsReady()) {
+        LoggerProxy.logger.log(
+          'Reachability:index#startTimers --> Reachability checks timed out (VMN timeout)'
+        );
+
+        this.resolveReachabilityPromise();
+      }
+    }, VIDEO_MESH_TIMEOUT * 1000);
+
+    this.publicCloudTimer = setTimeout(() => {
+      this.publicCloudTimer = undefined;
+
+      LoggerProxy.logger.log(
+        `Reachability:index#startTimers --> Reachability checks timed out (${DEFAULT_TIMEOUT}s)`
+      );
+
+      // resolve the promise, so that the client won't be blocked waiting on meetings.register() for too long
+      this.resolveReachabilityPromise();
+    }, DEFAULT_TIMEOUT * 1000);
+
+    this.overallTimer = setTimeout(() => {
+      this.overallTimer = undefined;
+      this.abortClusterReachability();
+      this.emit(
+        {
+          file: 'reachability',
+          function: 'overallTimer timeout',
+        },
+        'reachability:done',
+        {}
+      );
+      this.sendMetric();
+
+      LoggerProxy.logger.log(
+        `Reachability:index#startTimers --> Reachability checks fully timed out (${OVERALL_TIMEOUT}s)`
+      );
+    }, OVERALL_TIMEOUT * 1000);
+  }
+
+  /**
+   * Stores given reachability results in local storage
+   *
+   * @param {ReachabilityResults} results
+   * @returns {Promise<void>}
+   */
+  private async storeResults(results: ReachabilityResults) {
```

Comment:
SDK has always been storing it in local storage, I haven't changed it. I guess the idea was to have results ready to use in case we reload the page and don't do reachability before joining a meeting, but right now reachability is always done as part of `webex.meetings.register()` call that you have to make before being able to join anything, so there is not much benefit of using local storage. We might want to re-evaluate this, but it's out of scope for this PR.